lib/client.js

```diff
@@ -154,22 +220,29 @@ module.exports = function (dependencies) {
         resolve({ device, error });
       });
 
+      if (logger.enabled) {
+        this.session.on("close", () => {
+          logger("Request closed");
+        });
+
+        this.session.on("frameError", (frameType, errorCode, streamId) => {
+          logger(`Request frame error: (frameType: ${frameType}, errorCode ${errorCode}, streamId: ${streamId})`);
+        });
+      }
+
       request.end();
     });
   };
 
   Client.prototype.shutdown = function shutdown(callback) {
+    if (logger.enabled) {
+      logger('Called client.shutdown()');
+    }
+    this.isDestroyed = true;
     if (this.healthCheckInterval) {
       clearInterval(this.healthCheckInterval);
```

Comment:
nit: could set healthCheckInterval to null after clearing the interval

Reply:
done